src/omnisharp/OmnisharpDownloader.ts

```diff
@@ -4,69 +4,96 @@
  *--------------------------------------------------------------------------------------------*/
 
 import * as vscode from 'vscode';
-import { PackageManager, Package } from '../packages';
+import { PackageManager, Package, Status } from '../packages';
 import { PlatformInformation } from '../platform';
 import { Logger } from '../logger';
 import TelemetryReporter from 'vscode-extension-telemetry';
-import { GetPackagesFromVersion } from './OmnisharpPackageCreator';
-import { GetDependenciesAndDownloadPackages, SetStatus, GetAndLogPlatformInformation, ReportInstallationError, SendInstallationTelemetry } from '../OmnisharpDownload.Helper';
+import { GetPackagesFromVersion, GetVersionFilePackage } from './OmnisharpPackageCreator';
+import { SetStatus, LogPlatformInformation, ReportInstallationError, SendInstallationTelemetry, GetNetworkDependencies } from '../OmnisharpDownload.Helper';
 
 export class OmnisharpDownloader {
+    private status: Status;
+    private statusItem: vscode.StatusBarItem;
+    private proxy: string;
+    private strictSSL: boolean;
+    private packageManager: PackageManager;
+    private telemetryProps: any;
+
     public constructor(
         private channel: vscode.OutputChannel,
         private logger: Logger,
         private packageJSON: any,
+        private platformInfo: PlatformInformation,
         private reporter?: TelemetryReporter) {
+
+        let statusObject = SetStatus();
+        this.status = statusObject.Status;
+        this.statusItem = statusObject.StatusItem;
+
+        let networkObject = GetNetworkDependencies();
+        this.proxy = networkObject.Proxy;
+        this.strictSSL = networkObject.StrictSSL;
+
+        this.telemetryProps = {};
```

Comment:
What is this empty object for?

Reply:
https://github.com/OmniSharp/omnisharp-vscode/blob/experiment_download/src/OmnisharpDownload.Helper.ts#L47

This is used for the logging. This initialisation is the same as in the CSharpExtDownloader - https://github.com/OmniSharp/omnisharp-vscode/blob/experiment_download/src/CSharpExtDownloader.ts#L52